### PR TITLE
repairing berkeley testnet module compatibility

### PR DESCRIPTION
**Pull request by stevenplatt**
Explain your changes:
* This PR updates variables references within the following Terraform modules: 
`mina/automation/terraform/modules/kubernetes`
`mina/automation/terraform/modules/o1-testnet`

* Code used for Lucy testnet deployments had previously been refactored. This refactor broke compatibility with testnets being deployed outside of Lucy (for testnets such as `berkeley` and `testworld-2-0`). This was due to the referenced variable name changes. 

* This PR restores compatibility to the impacted terraform modules, so that they can be used both for Lucy and non-Lucy deployments. 

Explain how you tested your changes:
* To test that this change allows deployments outside of Lucy, a test deployment of the `berkeley` testnet was used done. This did not work initially, due to the variable name changes, but allowed iteratively editing the deploy files until the push worked correctly. This PR is the final version of working code for manual testnet deployments. 

Buildkite link: https://buildkite.com/o-1-labs-2/mina-end-to-end-nightlies/builds/813

* To test that this code does not break compatibility for Lucy testnet deployments, the same code was run through the o1 nightly CI pipeline to verify Lucy testnet deploys still worked.

Buildkite link: https://buildkite.com/o-1-labs-2/mina-o-1-labs/builds/31609

Checklist:

- [x] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [NA] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [x] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [NA] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [NA] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

### CODE REVIEW

File: automation/terraform/modules/kubernetes/testnet/variables.tf [hunk 1]
```diff
@@ -203,15 +203,15 @@ variable "seed_external_port" {
...
   type = list(
     object({
-      name               = string,
-      class              = string,
-      libp2p_secret      = string,
+      name             = string,
+      class            = string,
+      libp2p_secret    = string,
       libp2p_secret_pw = string
       # external_port      = number,
```
Comment (dkijania) [inline automation/terraform/modules/kubernetes/testnet/variables.tf:210]: Do you have any plans to remove those comments. They are located not only in this file but across all terraform sources? I wonder if they were only leftovers ?

File: automation/terraform/modules/o1-testnet/testnet.tf [hunk 2]
```diff
@@ -52,65 +52,67 @@ module "kubernetes_testnet" {
...
   snark_coordinators = var.snark_coordinators
 
-  block_producer_key_pass = var.block_producer_key_pass
-  block_producer_configs = [for i, bp in local.block_producer_configs:
+  # block_producer_key_pass = var.block_producer_key_pass
+  block_producer_configs = [for i, bp in local.block_producer_configs :
     {
-      name                   = bp.name
-      class                  = bp.class
+      name  = bp.name
```
Comment (dkijania) [inline automation/terraform/modules/o1-testnet/testnet.tf:69]: Did you ran some formatter tool or manually changed tabs? It looks like there is some inconsistency in tabbing :)

File: automation/terraform/testnets/berkeley/main.tf [hunk 3]
```diff
@@ -94,7 +94,7 @@ module "berkeley" {
   watchdog_image              = "gcr.io/o1labs-192920/watchdog:0.4.13"
   use_embedded_runtime_config = true
 
-  archive_node_count            = 3
+  archive_node_count            = 2
   mina_archive_schema           = "create_schema.sql"
   mina_archive_schema_aux_files = ["https://raw.githubusercontent.com/MinaProtocol/mina/8468f29424fa5e70cfedf69ee281f4fa24037dfe/src/app/archive/create_schema.sql", "https://raw.githubusercontent.com/MinaProtocol/mina/8468f29424fa5e70cfedf69ee281f4fa24037dfe/src/app/archive/zkapp_tables.sql"]
```
Comment (dkijania) [inline automation/terraform/testnets/berkeley/main.tf:99]: I know that it wasn't your change but I'm a bit worried to have hardcoded path to commit. Do you have plan to use some placeholder ? or fail if we won't override those variables?

**Comment by dkijania** [comment]
overall LGTM

**Comment by stevenplatt** [comment]
!ci-build-me

**Comment by dkijania** [comment]
I'm referring to :

https://github.com/MinaProtocol/mina/blob/berkeley/buildkite/src/Jobs/Test/TerraformNetworkTest.dhall test

**Comment by stevenplatt** [comment]
> I'm referring to :
> 
> https://github.com/MinaProtocol/mina/blob/berkeley/buildkite/src/Jobs/Test/TerraformNetworkTest.dhall test

I tested it in my repo, and can confirm that it would not be fixed. It will need the additional values added to `main.tf`. This is the same change I will do when merging the berkeley and testworld configs. I can update this file as well, but it would be a separate PR. This PR is only to bring the modules into sync, then the `main.tf` files of the testnets as a step-2.

**Comment by stevenplatt** [comment]
The following is the error generated currently, by the existing `main.tf` files: 

```
Error: Invalid value for module argument

  on main.tf line 109, in module "ci_testnet":
 109:   snark_coordinators = [
 110:     {
 111:       snark_worker_replicas   = var.snark_worker_count
 112:       snark_worker_fee        = "0.025"
 113:       snark_worker_public_key = "B62qk4nuKn2U5kb4dnZiUwXeRNtP1LncekdAKddnd1Ze8cWZnjWpmMU"
 114:       snark_coordinators_host_port  = 10401
 115:     }
 116:   ]

The given value is not suitable for child module variable "snark_coordinators"
defined at ../../modules/o1-testnet/inputs.tf:192,1-30: element 0: attributes
"persist_working_dir" and "snark_coordinator_name" are required.
```

It is possible that additional values need to be updated, but I would not see these until iterating and trying `terraform plan` again.

**Pull request merged**